control_protocol.md

```diff
@@ -0,0 +1,500 @@
+
+**TODO**: Add messages for this based on conclusion to [Issue #21](issues/21).
+
+#### Presentation Display Availablity Response
+
+The presentation display shall send a response for each Presentation Display
+Availability Request as follows.
+
+```
+Flavor:  Response
+Type:    0x0001
+Subtype: 0x0002
+
+Byte Offset
+  40           +-----------------------+
+               + NUM_URLS              +
+  42           +-----------------------+
+               + AVAILABILITY_RESULT_1 +
+  43           +-----------------------+
+               .                       .
+               +-----------------------+
+               + AVAILABILITY_RESULT_N +
+               +-----------------------+
+```
+
+- `NUM_URLS` is an unsigned positive 16-bit integer that is identical to
+  `NUM_URLS` in the corresponding Presentation Display Availability Request.
+- Each `AVAILABILITY_RESULT` is an unsigned 8-bit integer containing the
+  availability result for the Nth URL in the request.  The results are as follows:
+  
+Availability Result | Meaning
+--------------------|--------
+0                   | The URL is not compatible with the display.
+1                   | The URL is compatible with the display.
+10                  | The URL was not valid URL.
+100                 | The availability check timed out.
+101                 | The availability check failed (transient error).
+102                 | The availability check failed (permanent error).
+199                 | Unknown or other error processing URL.
+
+
+### Presentation Intiation Request
+
+This message is sent by a controller to start presentation on a presentation
+display.  The URL must be one that has been previously reported as compatible by
+that display.
+```
+Flavor:  Request
+Type:    0x0001
+Subtype: 0x0003
+
+Byte Offset
+  32           +-----------------------+
+               +  PRESENTATION_ID      +
+  160          +-----------------------+
+               +  ACCEPT_LANGUAGE      +
+  288          +-----------------------+
+               +  URL_LENGTH           +
+  292          +-----------------------+
+               +  URL_CONTENT          +
+               +-----------------------+
+```
+
+- `PRESENTATION_ID` is a null (zero-byte) terminated ASCII string of 128 bytes
+  that communicates the ID for the presentation.  Values shorter than 128 bytes
+  should be zero-byte padded.
+- `ACCEPT_LANGUAGE` is a null (zero-byte) terminated ASCII string of 128 bytes
+  that communicates the preferred locale for the presentation.  It is in the
+  format defined by the `Accept-Language:` header for HTTP, defined
+  in [RFC 7231](https://tools.ietf.org/html/rfc7231#section-5.3.5).
+  Values shorter than 128 bytes should be zero-byte padded.
+- `URL_LENGTH` is an unsigned positive 32-bit integer with the length, in bytes,
+  of the presentation URL.
+- `URL_CONTENT` is the presentation URL, encoded according to RFC 3986.
```

Comment:
Do we want to allow arbitrary HTTP header to be override, i.e. make receiver send particular cookie as sender to simplify the authentication procedure?

Reply:
That seems like a reasonable enhancement, although the API spec doesn't call for it now.  Done.